mod/reportedcontent/classes/Elgg/ReportedContent/Notifications/CreateReportedContentNotificationHandler.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Elgg\ReportedContent\Notifications;
+
+use Elgg\Notifications\NotificationEventHandler;
+use Elgg\Database\QueryBuilder;
+
+class CreateReportedContentNotificationHandler extends NotificationEventHandler {
+	
+	/**
+	 * {@inheritDoc}
+	 */
+	public function getSubscriptions(): array {
+		$admins = elgg_get_entities([
+			'type' => 'user',
+			'limit' => false,
+			'batch' => true,
+			'metadata_name_value_pairs' => [
+				[
+					'name' => 'admin',
+					'value' => 'yes',
+				],
+			],
+			'private_setting_name_value_pairs' => [
+				[
+					'name' => 'admin_notification',
+					'value' => 1,
+				],
+			],
+			'private_setting_name_prefix' => 'plugin:user_setting:reportedcontent:',
+			'wheres' => [
+				function (QueryBuilder $qb, $main_alias) {
+					// no idea why an admin would report content and not fix it themselfs, but just in case
+					return $qb->compare("{$main_alias}.guid", '!=', $this->event->getActorGUID(), ELGG_VALUE_GUID);
+				}
+			],
+		]);
+		
+		$subscriptions = [];
+		foreach ($admins as $admin) {
+			$subscriptions[$admin->guid] = ['email'];
+		}
+		
+		return $subscriptions;
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 */
+	public static function isConfigurableByUser(): bool {
+		return false;
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function getNotificationSubject(\ElggUser $recipient, string $method): string {
+		return elgg_echo('reportedcontent:notifications:create:admin:subject', [], $recipient->getLanguage());
```

Comment:
is this needed? wasn't there magic?

Reply:
yes, but gets more params (yes you could ignore those. this is more explicit